include/mem_host.h

```diff
@@ -21,105 +21,185 @@
 #ifndef DOSBOX_MEM_HOST_H
 #define DOSBOX_MEM_HOST_H
 
+/* These functions read and write fixed-width unsigned integers to/from
+ * byte-arrays using DOS/little-endian byte ordering.
+ *
+ * Values returned or passed to these functions are expected to be integers
+ * using native endianess representation for the host machine.
+ *
+ * They are safe to use even when byte-array address is not aligned according
+ * to desired integer width.
+ */
+
 #include <cstdint>
 
 #include "byteorder.h"
 #include "mem_unaligned.h"
 
-/*  These functions read and write 16, 32, and 64 bit uint's to
- *  and from unaligned 8-bit memory in DOS/little-endian ordering.
+/* Use host_read* functions to replace endian branching and byte swapping code,
+ * such as:
+ *
+ *   #ifdef WORDS_BIGENDIAN
+ *   auto x = byteswap(*(uint16_t*)(arr));
+ *   #else
+ *   auto x = *(uint16_t*)(arr);
+ *   #endif
  */
 
-// Read and write single-byte values
-constexpr uint8_t host_readb(const uint8_t *var)
+// Read a single-byte value; provided for backwards-compatibility only.
+constexpr uint8_t host_readb(const uint8_t *arr) noexcept
 {
-	return *var;
+	return *arr;
 }
 
-static inline void host_writeb(uint8_t *var, const uint8_t val)
+// Read a 16-bit word from 8-bit DOS/little-endian byte-ordered memory.
+static inline uint16_t host_readw(const uint8_t *arr) noexcept
 {
-	*var = val;
+	return le16_to_host(read_unaligned_uint16(arr));
 }
 
-/*  Read a 16-bit WORD from 8-bit DOS/little-endian byte-ordered memory.
- *  Use this instead of endian branching and byte swapping, such as:
- *  #if BIG_ENDIAN byteswap(*(uint16_t*)(arr)) #else *(uint16_t*)(arr) 
- */
-static inline uint16_t host_readw(const uint8_t *arr)
+// Read a 32-bit double-word from 8-bit DOS/little-endian byte-ordered memory.
+static inline uint32_t host_readd(const uint8_t *arr) noexcept
 {
-	return le16_to_host(read_unaligned_uint16(arr));
+	return le32_to_host(read_unaligned_uint32(arr));
 }
 
-/*  Read an array-indexed 16-bit WORD from 8-bit DOS/little-endian byte-ordered
- *  memory. Use this instead of endian branching and byte swapping, such as:
- *  #if BIG_ENDIAN byteswap(((uint16_t*)arr)[i]) #else ((uint16_t*)arr)[i]
+// Read a 64-bit quad-word from 8-bit DOS/little-endian byte-ordered memory.
+static inline uint64_t host_readq(const uint8_t *arr) noexcept
+{
+	return le64_to_host(read_unaligned_uint64(arr));
+}
+
+/* Use host_read*_at functions to replace endian branching and byte swapping
+ * code such as:
+ *
+ *   #ifdef WORDS_BIGENDIAN
+ *   auto x = byteswap(((uint16_t*)arr)[i]);
+ *   #else
+ *   auto x = ((uint16_t*)arr)[i];
```

Comment:
Suggest using `[idx]` in the examples to match with the function argument `idx`.